packages/vite/src/node/plugins/html.ts

```diff
@@ -330,7 +330,7 @@ export function buildHtmlPlugin(config: ResolvedConfig): Plugin {
     async transform(html, id) {
       if (id.endsWith('.html')) {
         id = normalizePath(id)
-        const relativeUrlPath = path.posix.relative(config.root, id)
+        const relativeUrlPath = path.relative(config.root, id)
```

Comment:
It should be `.posix` in both
```suggestion
        const relativeUrlPath = path.posix.relative(config.root, id)
```

Reply:
Using `.posix` here will cause path parsing errors on Windows platforms.
```ts
path.posix.relative('c:/a', 'C:/a/index.html') // => '../../C:/a/index.html'
path.relative('c:/a',  'C:/a/index.html') // => 'index.html'
```